docs: document DnD related part names (#3578)
vaadin/docs: articles/components/grid/styling.adoc

Row with open details:: `vaadin-grid+++<wbr>+++**::part(details-open-row)**`

> === Drag and Drop

Reviewer: ```suggestion
=== Drag & Drop
```

Author: Done

The following terminology is used in the description of these part names:

> - Ghost: the semi-transparent copy of the row that is dragged around by the pointer
> - Drag-source: the non-moving rows that are being dragged
> - Drop-target: a row or a gap between rows that the ghost can be dropped onto

Reviewer: ```suggestion
Drag source:: The row that is being dragged by the user. This part doesn't visually move during the drag.

Ghost:: A copy of the drag source (i.e., dragged row), which follows the user's pointer during the drag operation.

Drop target:: A row or a space between rows that the ghost can be dropped onto.
```

Author: I'm not sure it's a good idea to make the lists appear the same as they convey very different information. The explanations aren't part names so they should not appear as such.

Reviewer: That’s a good point. I changed the suggestion back to a regular list, but made the terms bold for emphasis.

> &#10;

Reviewer: I assume this is here to separate the two lists. Let’s use what the AsciiDoc documentation suggests: https://docs.asciidoctor.org/asciidoc/latest/lists/separating/

```suggestion
// separate lists
```

Author: Done

> Cell in the row being dragged (the “ghost”):: `vaadin-grid+++<wbr>+++**::part(dragstart-row-cell)**`
> Cell in a drag-source row:: `vaadin-grid+++<wbr>+++**::part(drag-source-row-cell)**`
> Cell in a row that ghost hovers above, with drop-target above:: `vaadin-grid+++<wbr>+++**::part(dragover-above-row-cell)**`
> Cell in a row that ghost hovers above, with drop-target below:: `vaadin-grid+++<wbr>+++**::part(dragover-below-row-cell)**`
> Cell in a drop-target row that ghost hovers above:: `vaadin-grid+++<wbr>+++**::part(dragover-on-top-row-cell)**`
> Cell in a row that isn’t draggable:: `vaadin-grid+++<wbr>+++**::part(drag-disabled-row-cell)**`
> Cell in a row that isn’t a drop-target:: `vaadin-grid+++<wbr>+++**::part(drop-disabled-row-cell)**`
> Row being dragged (the “ghost”):: `vaadin-grid+++<wbr>+++**::part(dragstart-row)**`
> Row that ghost hovers above, with drop-target above:: `vaadin-grid+++<wbr>+++**::part(dragover-above-row)**`
> Row that ghost hovers above, with drop-target below:: `vaadin-grid+++<wbr>+++**::part(dragover-below-row)**`
> Drop-target row that ghost hovers above:: `vaadin-grid+++<wbr>+++**::part(dragover-on-top-row)**`
> A row that isn’t draggable:: `vaadin-grid+++<wbr>+++**::part(drag-disabled-row)**`
> A row that isn’t a drop-target :: `vaadin-grid+++<wbr>+++**::part(drop-disabled-row)**`

Reviewer: I tried to make this more understandable, but I’m not sure I succeeded.

```suggestion
The cells in the “ghost” row::
`vaadin-grid+++<wbr>+++**::part(dragstart-row-cell)**`

The cells in the drag source row::
`vaadin-grid+++<wbr>+++**::part(drag-source-row-cell)**`

The cells of a row when the drop target is between this row and the row before it (i.e., above this row)::
`vaadin-grid+++<wbr>+++**::part(dragover-above-row-cell)**`

The cells of a row when the drop target is between this row and the row after it (i.e., below this row)::
`vaadin-grid+++<wbr>+++**::part(dragover-below-row-cell)**`

The cells of a row when the drop target is the row itself (i.e., on top of this row)::
`vaadin-grid+++<wbr>+++**::part(dragover-on-top-row-cell)**`

The cells of a row that isn’t draggable::
`vaadin-grid+++<wbr>+++**::part(drag-disabled-row-cell)**`

The cells of a row that isn’t a drop target::
`vaadin-grid+++<wbr>+++**::part(drop-disabled-row-cell)**`

The dragged row (i.e., the “ghost”)::
`vaadin-grid+++<wbr>+++**::part(dragstart-row)**`

The row which is the drop target, and the target is between this row and the row before it (i.e., above this row)::
`vaadin-grid+++<wbr>+++**::part(dragover-above-row)**`

The row which is the drop target, and the target is between this row and the row after it (i.e., below this row)::
`vaadin-grid+++<wbr>+++**::part(dragover-below-row)**`

The row which is the drop target, and the target is the row itself (i.e., on top of this row)::
`vaadin-grid+++<wbr>+++**::part(dragover-on-top-row)**`

A row that isn’t draggable::
`vaadin-grid+++<wbr>+++**::part(drag-disabled-row)**`

A row that isn’t a drop target ::
`vaadin-grid+++<wbr>+++**::part(drop-disabled-row)**`
```

Author: Yeah, the part names are hard to describe,. This PR is the result of Rolf's feedback put on top of my (someone fairly new to the grid) understanding of the parts.
